tools/topology/sof-tgl-max98373-rt5682.m4

```diff
@@ -37,12 +37,21 @@ DEBUG_START
 # PCM99 <---- volume <---- DMIC01 (dmic 48k capture)
 # PCM100 <---- kpb <---- DMIC16K (dmic 16k capture)
 
+
+ifdef(`AMP_SSP',`',`errprint(note: Define AMP_SSP for speaker amp SSP Index)')
 # Smart amplifier related
 # SSP related
 #define smart amplifier SSP index
-define(`SMART_SSP_INDEX', 1)
+define(`SMART_SSP_INDEX', AMP_SSP)
 #define SSP BE dai_link name
+ifelse(AMP_SSP, `1',
+`
 define(`SMART_SSP_NAME', `SSP1-Codec')
+', `')
+ifelse(AMP_SSP, `2',
+`
+define(`SMART_SSP_NAME', `SSP2-Codec')
+', `')
```

Comment:
We can use concat then no ifelse needed, something like:
define(`SMART_SSP_NAME', concat(concat(`SSP', AMP_SSP), `-Codec'))

Reply:
oh string concatenation is there. Thank you @keyonjie .
Let me try and update.

Reply:
> We can use concat then no ifelse needed, something like:
> 
> ```
> define(`SMART_SSP_NAME', concat(concat(`SSP', AMP_SSP), `-Codec'))
> ```

Done, updated.